install_jupyter_kernel.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+# Copyright 2021-2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import shutil
+import sys
+
+from jupyter_client.kernelspec import KernelSpecManager
+
+
+def find_python_module(legate_dir):
+    lib_dir = os.path.join(legate_dir, "lib")
+    python_lib = None
+    for f in os.listdir(lib_dir):
+        if f.startswith("python") and not os.path.isfile(f):
+            if python_lib is not None:
+                print(
+                    "WARNING: Found multiple python supports in your Legion "
+                    "installation."
+                )
+                print("Using the following one: " + str(python_lib))
+            else:
+                python_lib = os.path.join(
+                    lib_dir, os.path.join(f, "site-packages")
+                )
+
+    if python_lib is None:
+        raise Exception("Cannot find a Legate python library")
+    return python_lib
+
+
+if __name__ == "__main__":
+    (script_name, extension) = os.path.splitext(__file__)
+    if extension == ".py":
+        # running (source_dir)/install_jupyter_kernel.py
+        try:
+            legate_dir = os.environ["LEGATE_DIR"]
+            legate_dir = os.path.abspath(legate_dir)
+        except KeyError:
+            print(
+                "Please specify the legate installation dir "
+                "by setting LEGATE_DIR"
+            )
+            sys.exit(1)
+    else:
+        # running /LEGATE_DIR/bin/install_jupyter_kernel
+        legate_bin_dir = os.path.dirname(os.path.realpath(__file__))
+        legate_dir = os.path.abspath(os.path.join(legate_bin_dir, os.pardir))
```

Comment:
I don't like the script changing behavior based on its extension. I would just assume that the script has been copied into the installation directory, and just keep the two last lines.

Reply:
fixed.